mail_outbound_static/tests/test_ir_mail_server.py

```diff
@@ -0,0 +1,369 @@
+        cls._delete_mail_servers()
+        cls.IrMailServer.create(
+            {
+                "name": "localhost",
+                "smtp_host": "localhost",
+                "smtp_from": cls.email_from,
+            }
+        )
+        message_file = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), "test.msg"
+        )
+        with open(message_file, "r") as fh:
+            cls.message = message_from_string(fh.read())
+
+    @classmethod
+    def _delete_mail_servers(cls):
+        """Delete all available mail servers"""
+        all_mail_servers = cls.IrMailServer.search([])
+        if all_mail_servers:
+            all_mail_servers.unlink()
+
+    def _init_mail_server_domain_whilelist_based(self):
+        self._delete_mail_servers()
+        self.assertFalse(self.IrMailServer.search([]))
+        self.mail_server_domainone = self.IrMailServer.create(
+            {
+                "name": "sandbox domainone",
+                "smtp_host": "localhost",
+                "smtp_from": "notifications@domainone.com",
+                "domain_whitelist": "domainone.com",
+            }
+        )
+        self.mail_server_domaintwo = self.IrMailServer.create(
+            {
+                "name": "sandbox domaintwo",
+                "smtp_host": "localhost",
+                "smtp_from": "hola@domaintwo.com",
+                "domain_whitelist": "domaintwo.com",
+            }
+        )
+        self.mail_server_domainthree = self.IrMailServer.create(
+            {
+                "name": "sandbox domainthree",
+                "smtp_host": "localhost",
+                "smtp_from": "notifications@domainthree.com",
+                "domain_whitelist": "domainthree.com,domainmulti.com",
+            }
+        )
+
+    def _skip_test(self, reason):
+        _logger.warn(reason)
+        self.skipTest(reason)
+
+    def _send_mail(
+        self,
+        message,
+        mail_server_id=None,
+        smtp_server=None,
+        smtp_port=None,
+        smtp_user=None,
+        smtp_password=None,
+        smtp_encryption=None,
+        smtp_ssl_certificate=None,
+        smtp_ssl_private_key=None,
+        smtp_debug=False,
+        smtp_session=None,
+    ):
+        smtp = smtp_session
+        if not smtp:
+            smtp = self.IrMailServer.connect(
+                smtp_server,
+                smtp_port,
+                smtp_user,
+                smtp_password,
+                smtp_encryption,
+                smtp_from=message["From"],
+                ssl_certificate=smtp_ssl_certificate,
+                ssl_private_key=smtp_ssl_private_key,
+                smtp_debug=smtp_debug,
+                mail_server_id=mail_server_id,
+            )
+
+        send_from, send_to, message_string = self.IrMailServer._prepare_email_message(
+            message, smtp
+        )
+        self.IrMailServer.send_email(message)
+        return message_string
```

Comment:
this is needed, just because, I'm not an expert with email servers and couldn't find any other idea how to get `smtp_seesion` parameter to pass directly to `_prepare_email_message` method of `ir.mail_server`.
Would really appreciate any guidance. Thanks